packages/server-wallet/src/wallet/store.ts

```diff
@@ -632,103 +599,75 @@ class StoreError extends WalletError {
   }
 }
 
+/**
+ * Construct a new Channel and insert it into the database, ensuring that the Nonce
+ * provided for this new channel is also the newest we have ever seen.
+ */
 async function createChannel(
   constants: ChannelConstants,
   fundingStrategy: FundingStrategy,
   fundingLedgerChannelId: Address | undefined,
   txOrKnex: TransactionOrKnex
 ): Promise<Channel> {
-  const channelId = calculateChannelId(constants);
+  const addresses = constants.participants.map(p => p.signingAddress);
+
+  const signingWallet = await SigningWallet.query(txOrKnex)
+    .whereIn('address', addresses)
+    .first();
 
-  const {address: signingAddress} = await getSigningWallet(constants, txOrKnex);
+  if (!signingWallet) throw new StoreError(StoreError.reasons.notInChannel);
 
-  const cols: RequiredColumns = pick(
-    {
-      ...constants,
-      channelId,
-      vars: [],
-      chainServiceRequests: [],
-      signingAddress,
-      fundingStrategy,
-    },
-    ...CHANNEL_COLUMNS
+  await Nonce.ensureLatest(constants.channelNonce, addresses, txOrKnex);
```

Comment:
This is _new_ behaviour, if I am not mistaken, and not just a refactor?

Reply:
Same behaviour as before! It used to be a bit more hidden